multichain-aggregator/multichain-aggregator-logic/src/repository/counters.rs

```diff
@@ -0,0 +1,68 @@
+use crate::types::counters::ChainCounters;
+use chrono::{NaiveDateTime, NaiveTime, Utc};
+use entity::counters_global_imported::{
+    ActiveModel as GlobalCountersActiveModel, Column as GlobalCountersColumn,
+    Entity as GlobalCountersEntity,
+};
+use sea_orm::{
+    ActiveModelTrait,
+    ActiveValue::{NotSet, Set},
+    ColumnTrait, ConnectionTrait, DbErr, EntityTrait, QueryFilter, QueryOrder,
+};
+
+pub async fn upsert_chain_counters<C>(db: &C, data: ChainCounters) -> Result<(), DbErr>
+where
+    C: ConnectionTrait,
+{
+    let ChainCounters {
+        chain_id,
+        timestamp,
+        daily_transactions_number,
+        total_transactions_number,
+        total_addresses_number,
+    } = data;
+
+    let day_start = NaiveDateTime::new(timestamp.date(), NaiveTime::from_hms_opt(0, 0, 0).unwrap());
+    let day_end = day_start + chrono::Duration::days(1);
+
+    let existing = GlobalCountersEntity::find()
+        .filter(GlobalCountersColumn::ChainId.eq(chain_id))
+        .filter(GlobalCountersColumn::UpdatedAt.gte(day_start))
+        .filter(GlobalCountersColumn::UpdatedAt.lt(day_end))
+        .order_by_desc(GlobalCountersColumn::UpdatedAt)
+        .one(db)
+        .await?;
```

Comment:
What do you think about creating an index like this?
`CREATE UNIQUE INDEX counters_global_imported_unique_day_updated_at_index ON counters_global_imported (date_trunc('day', updated_at));`

Then we can use this index as `ON CONFLICT` target to upsert counters in just one db operation.

Reply:
Or we can explicitly add a `date` column as suggested above
https://github.com/blockscout/blockscout-rs/pull/1446#discussion_r2208551095

Reply:
Yes, I decided to introduce the date field. I think it's a more natural approach. Also I've added (chain_id, date) index for this table to improve DB performance. https://github.com/blockscout/blockscout-rs/pull/1446/commits/9f43b27b4258eaad0428acbf0108f58b5a1da940